DXMainClient/Domain/Multiplayer/TeamStartMapping.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ClientCore;
+using Newtonsoft.Json;
+
+namespace DTAClient.Domain.Multiplayer
+{
+    public class TeamStartMapping
+    {
+        private const char SEPARATOR = ':';
+        private const char LIST_SEPARATOR = ',';
+        private const string INVALID_TEAM_MAPPING_ERROR = "Invalid team mapping \"{0}\"";
+        
+        public const string BLOCK = "-";
+        public static readonly List<string> TEAMS = new List<string>() { BLOCK }.Concat(ProgramConstants.TEAMS).ToList();
+
+        [JsonProperty("t")]
+        public string Team { get; set; }
+
+        [JsonProperty("s")]
+        public int Start { get; set; }
+
+        [JsonIgnore]
+        public bool IsValid => TeamIndex != -1;
+
+        [JsonIgnore]
+        public int TeamIndex => TEAMS.IndexOf(Team);
+
+        [JsonIgnore]
+        public int StartingWaypoint => Start - 1;
+
+        [JsonIgnore]
+        public bool IsBlock => Team == BLOCK;
+
+        /// <summary>
+        /// We write these out in the form A:1 where A is the team and 1 is the starting location on the map
+        /// as it would appear in the map preview.
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            return Team;
+        }
+
+        /// <summary>
+        /// Parse the incoming string into a <see cref="TeamStartMapping"/>.
+        /// This should be the exact opposite of the <see cref="ToString"/> method.
+        /// </summary>
+        /// <param name="str"></param>
+        /// <param name="index"></param>
+        /// <returns></returns>
+        private static TeamStartMapping FromString(string str, int index)
+        {
+            return new TeamStartMapping()
+            {
+                Team = str,
+                Start = index + 1
+            };
+        }
```

Comment:
Not sure if this is worth leaving in as a separate method as well, don't see it being useful outside of parsing function.

Reply:
The purpose of the `FromString` and `ToString` methods was to containing the serializing in/out of this class IN this class. This serializing occurs when reading in Map data and for sending/receiving these mappings via a IRC message.

Reply:
I get that, but do we need to split the stuff so much? I personally don't see a point, considering this is for internal use and it's used only once.